middlewares/checkIfUserAllowed.js

```diff
@@ -0,0 +1,73 @@
+const { ROLE_NAME, RESOURCE } = require("../constants/index.js");
+const Comment = require("../models/commentModel");
+const Post = require("../models/postModel.js");
+const User = require("../models/userModel.js");
+
+const checkIfUserAllowed = (resource) => async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    if (resource === RESOURCE.User) {
+      if (
+        req.method === "DELETE" ||
+        ("PUT" &&
+          (req.userRole === ROLE_NAME.ADMIN ||
+            req.userRole === ROLE_NAME.SUPERADMIN))
+      ) {
```

Comment:
```
 req.method === "DELETE" ||
        ("PUT" &&
          (req.userRole === ROLE_NAME.ADMIN ||
            req.userRole === ROLE_NAME.SUPERADMIN))
      ) {
```

If it is him deleting or updating his post or comments (it does'nt matter) what are you checking with code on lines 13-14, in addition, you check if request sender is an admin or superadmin and give permission in the code on lines 22-26